code/controllers/subsystem/daynightcycle.dm

```diff
@@ -0,0 +1,111 @@
+var/datum/subsystem/daynightcycle/SSDayNight
+var/list/daynight_turfs = list()
+/* Original Plan
+Morning	  - 2 Mins
+Sunrise   - 2 Mins
+Daytime   - 16 Minutes
+Afternoon - 16 Minutes
+Sunset    - 2 Minutes
+Nighttime - 36 Minutes
+*/
+					
+#define TOD_MORNING 	"#4d6f86" 
+#define TOD_SUNRISE 	"#fdc5a0"
+#define TOD_DAYTIME 	"#FFFFFF"
+#define TOD_AFTERNOON 	"#ffeedf"
+#define TOD_SUNSET 		"#75497e"
+#define TOD_NIGHTTIME 	"#001522"
+
+/datum/subsystem/daynightcycle
+	name          = "Day Night Cycle"
+	init_order    = SS_INIT_DAYNIGHT
+	display_order = SS_DISPLAY_DAYNIGHT
+	priority      = SS_PRIORITY_DAYNIGHT
+	wait          = 1 MINUTES
+	flags         = SS_FIRE_IN_LOBBY
+
+	var/current_timeOfDay = TOD_DAYTIME //We start tickers maxed out, and start on afternoon
+	var/next_light_power = 10
+	var/next_light_range = 1
+
+	//The initial values don't matter, it just needs to fire initially, then set itself into the cycle.
+	var/next_firetime = 0
+	var/list/currentrun
+
+/datum/subsystem/daynightcycle/New()
+	NEW_SS_GLOBAL(SSDayNight)
+
+/datum/subsystem/daynightcycle/Initialize()
+	get_turflist()
+	..()
+
+/datum/subsystem/daynightcycle/fire(resumed = FALSE)
+	if(flags & SS_NO_FIRE)
```

Comment:
`SS_NO_FIRE`'s intended purpose is to prevent a subsystem from being added to the processing list in the first place. Instead of keeping this check here, you can simply set `can_fire = FALSE`.
The correct approach would be to change the flags of the subsystems in `snaxi.dm`, which will only get compiled when the map is snaxi, and will require no runtime checks at all.
I'm aware this is also how the weather subsystem (and another one?) does things, and it works, but it's not correct.

Reply:
Aight, I redid it, does this look correct?

Reply:
Not quite. I imagined it would look something like this in snaxi.dm:
```
/datum/subsystem/daynightcycle
    flags = SS_FIRE_IN_LOBBY
```
with the default being something like this in daynightcycle.dm:
```
/datum/subsystem/daynightcycle
    flags = SS_NO_FIRE | SS_NO_INIT
```

so that maps would opt-in to having a day/night cycle at compile time

Reply:
Alright how about now, I did what I think you told me to do.

Reply:
Yep, works for me.

Reply:
I'd recommend testing it after this change if you haven't. Depending on compile order it may or may not work.